extract/extract/extractors/zalando.py

```diff
@@ -124,40 +128,33 @@ def extract_zalando(
         return None
 
 
-def get_sustainability(
-    beautiful_soup: BeautifulSoup, label_mapping: Dict[str, CertificateType]
-) -> List[str]:
+def get_sustainability(beautiful_soup: BeautifulSoup) -> Iterator[str]:
     """
     Extracts the sustainability information from HTML. Splash does not load all the information
     into the html anymore,so we have to extract the sustainability information from a JSON, stored
     inside a script tag.
 
     Args:
         beautiful_soup (BeautifulSoup): Parsed HTML
-        label_mapping (Dict) : Dictionary containing the mappings from Zalando's certificate names
-        to our CertificateTypes
 
     Returns:
-        List[str]: Ordered `list` of found sustainability labels
+        Iterator[str]: found sustainability labels
     """
 
-    # Load JSON holding the sustainability information
     data = json.loads(
         beautiful_soup.find("script", {"type": "application/json", "class": "re-1-13"}).get_text()
     )
-    labels = []
 
     # Loop over all nested items to find the JSON node holding the sustainability information
-    for key, item in data.get("graphqlCache", {}).items():
-        for entry in item.get("data", {}).get("product", {}).get("attributeSuperClusters", [{}]):
-            if entry.get("id", "") == "sustainability":
-                for cluster in entry.get("clusters", [{}]):
-                    if cluster.get("sustainabilityClusterKind", "") == "certificates":
-                        # Loop over all certificate attributes and add the label to the labels list
-                        for attribute in cluster.get("attributes", [{}]):
-                            labels.append(urllib.parse.unquote(attribute.get("label", "")))
-
-    if labels:
-        # Map the found labels to their respective CertificateType
-        return sorted({label_mapping.get(label, CertificateType.UNKNOWN) for label in labels})
-    return []
+    stack = [data]
+    while len(stack) > 0:
```

Comment:
Ok wow, took me a while but this is pretty nice! - could we rename `stack` and `node`? Maybe something like `nested_dicts` and `nested_dict`.

I like to iterate over lists where the list is plural and its items singular named, does this make sense to you?

Reply:
i changed it to use a for loop instead of the manual `.pop()`. and the names now match those used in the json spec.